### ISSUE TITLE
Ambiguous call when trying to call a stored function returning a table type

### ISSUE THREAD

**Issue by lukaseder** [cockroachdb/cockroach#93313]
**Describe the problem**

I can't call a stored function that returns a table type from `SELECT`

**To Reproduce**

```sql
create table t (i int, j int);
insert into t values (1, 2);

create or replace function f () returns t as 
$$
select i, j from t;
$$
language sql;

select f();
```

The error is:

> SQL Error [42725]: ERROR: ambiguous call: f(), candidates are:
> f() -> tuple{int AS i, int AS j}
> f() -> tuple{int AS i, int AS j}

But I'm not seeing the ambiguity in the catalog:

```sql
select count(*) from pg_proc where proname = 'f';
select count(*) from pg_proc where pronamespace = (
  select oid from pg_namespace where nspname = 'public'
);
```

Result for both queries:

```
|count|
|-----|
|1    |
```

Note, I also can't drop the function!

```sql
drop function f ();
```

> SQL Error [XXUUU]: ERROR: function name "f" is not unique

**Expected behavior**

I would expect this result as in native PostgreSQL:

```
|i  |j  |
|---|---|
|1  |2  |
```

**Environment:**
 - CockroachDB version: CockroachDB CCL v22.2.0 (x86_64-pc-linux-gnu, built 2022/12/05 16:37:36, go1.19.1)
 - Server OS: Linux (Docker)
 - Client app: Dbeaver (JDBC)

Jira issue: CRDB-22269

**Comment by mgartner** [comment]
I think this might be related to #93321. cc @DrewKimball

**Comment by DrewKimball** [comment]
I'm unable to reproduce this either on current master or on v22.2.0. @lukaseder do you have any other details from when you observed this?

**Comment by lukaseder** [comment]
Sure, I did this:

```
docker pull cockroachdb/cockroach
docker run -d --name=COCKROACHDB --hostname=localhost --net=roachnet -p 26257:26257 -p 8081:8080 -v "//c/data/cockroach" cockroachdb/cockroach start-single-node --ins
ecure
```

Then connected with Dbeaver:

![image](https://user-images.githubusercontent.com/734593/224287120-0e643da3-beb4-4484-b343-a268429c7995.png)

Then I ran this, as mentioned in the ticket:

```sql
create table t (i int, j int);
insert into t values (1, 2);

create or replace function f () returns t as 
$$
select i, j from t;
$$
language sql;

select f();
```

Result:

![image](https://user-images.githubusercontent.com/734593/224287210-8e4e8e00-52e4-4be8-8608-1db1df2760d9.png)

```sql
select version();
```

Produces:

> CockroachDB CCL v22.2.6 (x86_64-pc-linux-gnu, built 2023/03/03 19:31:13, go1.19.4)

Hope this helps

**Comment by lukaseder** [comment]
Not sure if it's relevant, but I'm running Docker Desktop version 4.14.1 (91661) on Windows:

```
C:\Users\lukas>ver

Microsoft Windows [Version 10.0.22621.1265]
```

**Comment by DrewKimball** [comment]
Thanks for the help, it reproduces for me using dbeaver.

**Comment by DrewKimball** [comment]
This is fixed by #96734, which will make it into the next 22.2 minor version.
